### PR TITLE
[#5401] Checksum API calculates hash using size in catalog (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: scripts/irods/test/test_ichksum.py [hunk 1]
```diff
@@ -378,3 +380,30 @@ def test_ichksum_groups_objects_appropriately_when_the_recursive_flag_is_set__is
...
+        # Make the catalog report the wrong size of the replica on disk.
+        # This change will cause icommands such as iget and istream to return an empty string.
+        self.admin.assert_icommand(['iadmin', 'modrepl', 'logical_path', data_object, 'replica_number', '0', 'DATA_SIZE', '0'])
+        self.admin.assert_icommand(['istream', 'read', data_object])
+
+        # Show that the calculated checksum matches the SHA256 checksum of an empty string.
+        replica_checksum = '47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='
+        self.admin.assert_icommand(['ichksum', '-f', data_object], 'STDOUT', ['sha2:' + replica_checksum])
+        self.assertEqual(replica_checksum, base64.b64encode(hashlib.sha256('').digest()))
+
```
Comment (trel) [inline scripts/irods/test/test_ichksum.py:396]: can we have an additional assertion with size 3 that checksum equals sha256('the')?

**Comment by korydraughn** [comment]
requires https://github.com/irods/irods_client_icommands/pull/194

File: server/core/include/finalize_utilities.hpp [hunk 2]
```diff
@@ -116,12 +116,13 @@ namespace irods
...
     /// \param[in/out] _comm
     /// \param[in] _fd l1 descriptor index
+    /// \param[in] _preserve_replica_state_table
     ///
     /// \returns int
     /// \retval return code of rs_replica_close
     ///
     /// \since 4.2.9
-    auto close_replica_without_catalog_update(RsComm& _comm, const int _fd) -> int;
+    auto close_replica_without_catalog_update(RsComm& _comm, const int _fd, bool _preserve_replica_state_table) -> int;
```
Comment (alanking) [inline server/core/include/finalize_utilities.hpp:125]: Could be `const bool`

**Comment by trel** [comment]
ready for # if you are

**Comment by korydraughn** [comment]
passed CI. added #

**Pull request merged**